src/epcpm/staticmodbustoc.py

```diff
@@ -0,0 +1,136 @@
+            )
+            h_file.write("\n#endif //__STATICMODBUS_INTERFACE_GEN_H__\n")
+
+        with self.c_path.open("w", newline="\n") as c_file:
+            c_file.write('#include "staticmodbusInterfaceGen.h"\n')
+            c_file.write('#include "interfaceGen.h"\n\n')
+            c_file.write("InterfaceItem_void * const staticmodbusAddrRegMap[] = \n{\n")
+
+            current_addr = 0
+            for member in self.wrapped.children:
+                builder = builders.wrap(
+                    wrapped=member,
+                    parameter_uuid_finder=self.parameter_uuid_finder,
+                    skip_sunspec=self.skip_sunspec,
+                )
+
+                addr_val, c_line = builder.gen()
+
+                # Fill the undefined registers to return NULL.
+                while current_addr < addr_val:
+                    null_line = f"    [{current_addr}] = NULL,\n"
+                    c_file.write(null_line)
+                    current_addr += 1
+                # Write the defined register that returns interfaceItem_<UUID>.
+                c_file.write(c_line)
+                # Update the current address for the next undefined register fill.
+                current_addr = addr_val + 1
+
+            c_file.write("};\n\n")
+
+
+@builders(epcpm.staticmodbusmodel.FunctionData)
+@attr.s
+class FunctionData:
+    wrapped = attr.ib()
+    parameter_uuid_finder = attr.ib()
+    skip_sunspec = attr.ib(default=False)
+
+    def gen(self):
+        uses_interface_item = False
+        if self.wrapped.parameter_uuid is not None:
+            parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
+            uses_interface_item = (
+                isinstance(parameter, epyqlib.pm.parametermodel.Parameter)
+                and parameter.uses_interface_item()
+            )
+
+        type_node = self.parameter_uuid_finder(self.wrapped.type_uuid)
+
+        # Generate the defined register that returns interfaceItem_<UUID> (or NULL for some cases).
+        if (
+            not self.skip_sunspec
+            and uses_interface_item
+            and not self.wrapped.not_implemented
+            and type_node is not None
+            and type_node.name != "staticmodbussf"
+        ):
+            # TODO: CAMPid 9685439641536675431653179671436
+            parameter_uuid = str(self.wrapped.parameter_uuid).replace("-", "_")
+            uuid_interface_val = f"&interfaceItem_{parameter_uuid}"
+        else:
+            uuid_interface_val = "NULL"
+        addr_val = self.wrapped.address
+        c_line = f"    [{addr_val}] = {uuid_interface_val},\n"
+
+        return addr_val, c_line
+
+
+@builders(epcpm.staticmodbusmodel.FunctionDataBitfield)
+@attr.s
+class FunctionDataBitfield:
+    wrapped = attr.ib()
+    parameter_uuid_finder = attr.ib()
+    skip_sunspec = attr.ib(default=False)
+
+    def gen(self):
+        # NULL for all FunctionDataBitfield objects.
```

Comment:
```suggestion
        # TODO: to be implemented, for now NULL for all FunctionDataBitfield objects
```

Reply:
Addressed in e91a3486cb2a1ece70f5257f1c457cbb94cdb5b0